core/chains/evm/logpoller/log_poller_test.go

```diff
@@ -2091,3 +2093,39 @@ func TestFindLCA(t *testing.T) {
 		})
 	}
 }
+
+func TestWhere(t *testing.T) {
+	address := common.HexToAddress("0x1234567890abcdef1234567890abcdef12345678")
+	eventSig := common.HexToHash("0xabcdef1234567890abcdef1234567890abcdef1234")
+	ts := time.Now()
+
+	expr1 := logpoller.NewAddressFilter(address)
+	expr2 := logpoller.NewEventSigFilter(eventSig)
+	expr3 := query.Timestamp(uint64(ts.Unix()), primitives.Gte)
+	expr4 := logpoller.NewConfirmationsFilter(evmtypes.Confirmations(0))
+
+	t.Run("Valid combination of filters", func(t *testing.T) {
+		result, err := logpoller.Where(expr1, expr2, expr3, expr4)
+		assert.NoError(t, err)
+		assert.Equal(t, []query.Expression{expr1, expr2, expr3, expr4}, result)
+	})
+
+	t.Run("No expressions (should return empty slice)", func(t *testing.T) {
+		result, err := logpoller.Where()
+		assert.NoError(t, err)
+		assert.Equal(t, []query.Expression{}, result)
+	})
+
+	t.Run("Invalid boolean expression", func(t *testing.T) {
```

Comment:
are the valid and invalid expressions documented somewhere? Would it make sense to go through all them in these tests and show how to properly use them?

Reply:
This `logpoller.Where()` function it's just a wrapper of the [original one](https://github.com/smartcontractkit/chainlink-common/blob/v0.2.1/pkg/types/query/query.go#L124) `query.Where()` where you don't need to provide a `key` param and just the expressions

The underlying function has extensive tests [here](https://github.com/smartcontractkit/chainlink-common/blob/v0.2.1/pkg/loop/internal/relayer/pluginprovider/chainreader/chain_reader_test.go#L250) which uses the test cases defined [here](https://github.com/smartcontractkit/chainlink-common/blob/v0.2.1/pkg/loop/internal/relayer/pluginprovider/chainreader/helper_test.go#L101). 

Let me know your thoughts if this seems enough